### ISSUE TITLE
Can't build pnet_packet dependency

### ISSUE THREAD

**Issue by kolen** [ewilken/hap-rs#12]
When building `hap = "0.0.3"` (and version from git master too) on MacOS, one of dependencies can't be built, with following error: 

```
   Compiling pnet_packet v0.21.0
error[E0599]: no method named `to_primitive_values` found for type `pnet_base::MacAddr` in the current scope
   --> /Users/kolen/items/pm_sensor/logging_node/target/debug/build/pnet_packet-3048cc3df534cf77/out/ethernet.rs:442:24
    |
442 |         let vals = val.to_primitive_values();
    |                        ^^^^^^^^^^^^^^^^^^^
```

Related issue in `pnet`: https://github.com/libpnet/libpnet/issues/361

Updating `pnet` dependency version to `0.22.0` (from `0.21.0`) fixes it. It builds and basic example from readme works (tried connecting with phone). Not thoroughly tested, however.

**Comment by ewilken** [comment]
Thanks for pointing out! This issue will be fixed together with an adoption of Rust 2018 and recent versions of Hyper & Tokio and a fix for thread safety (as requested in #5) that I'm hopefully gonna find time for releasing in the next few days.

**Comment by ewilken** [comment]
v0.0.4 is out with a fix.
